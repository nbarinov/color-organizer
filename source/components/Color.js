import React from 'react';
import PropTypes from 'prop-types';

import StarRating from './StarRating';

import '../style/color.css';

const Color = ({ className, title, color, rating = 0, onRemove = f => f, onRate = f => f }) => (
    <section className={(className) ? className + ' color' : 'color'}>
        <header className="color__header">
            <h1 className="color__title">{title}</h1>
            <button className="color__remove" onClick={onRemove} title="remove color">x</button>
        </header>
        <div className="color__block" style={{ backgroundColor: color }}></div>
        <footer className="color__footer">
            <StarRating starsSelected={rating} onRate={onRate} />
        </footer>
    </section>
);

Color.propTypes = {
    title: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    rating: PropTypes.number,
    onRemove: PropTypes.func,
    onRate: PropTypes.func,
};

export default Color