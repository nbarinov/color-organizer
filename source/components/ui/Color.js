import { Component } from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

import StarRating from './StarRating';
import TimeAgo from './TimeAgo';

import '../../style/color.css';

class Color extends Component {
    render() {
        const { className, id, title, color, rating, timestamp, onRate, onRemove, history } = this.props;
        
        return (
            <section className={(className) ? className + ' color' : 'color'}>
                <header className="color__header">
                    <h1 ref="title" className="color__title" onClick={() => history.push(`/${id}`)}>{title}</h1>
                    <button className="color__remove" onClick={onRemove} title="remove color">x</button>
                </header>
                <div className="color__block" style={{ backgroundColor: color }} onClick={() => history.push(`/${id}`)}></div>
                <footer className="color__footer">
                    <StarRating className="color__rating" starsSelected={rating} onRate={onRate} />
                    <TimeAgo className="color__timeago" timestamp={timestamp} />
                </footer>
            </section>
        );
    }
}

Color.propTypes = {
    title: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    rating: PropTypes.number,
    onRemove: PropTypes.func,
    onRate: PropTypes.func,
};

Color.defaultProps = {
    rating: 0,
    onRate: f => f,
    onRemove: f => f,
};

export default withRouter(Color);