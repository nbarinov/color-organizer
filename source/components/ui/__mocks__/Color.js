const ColorMock = ({ rating, onRate = f => f }) => (
    <div className="mock-color">
        <button className="rate" onClick={() => onRate(rating)} />
    </div>
);

ColorMock.displayName = 'ColorMock';

export default ColorMock;