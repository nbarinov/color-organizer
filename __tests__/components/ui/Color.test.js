import Color from '../../../source/components/ui/Color';

const { mount, shallow } = Enzyme;

describe("<Color /> UI Component", () => {
    describe("Rendering for UI", () => {
        it("Renders correct properties", () => {
            let output = shallow(
                <Color title="Test Color"
                    color="#f0f0f0"
                    rating={3}
                    timestamp="Mon Apr 11 2016 12:54:19 GMT-0700 (PDT)" />
            ).html();

            expect(output).toMatchSnapshot();
        });

        it("renders parental className color", () =>
            expect(
                shallow(
                    <Color title="Test Color"
                        color="#f0f0f0"
                        rating={3}
                        timestamp="Mon Apr 11 2016 12:54:19 GMT-0700 (PDT)"
                        className="parent__color" />
                )
                    .find('section.parent__color.color')
                    .length
            ).toBe(1)
        );
    });

    describe("Functions", () => {
        it("Invokes onRate property", () => {
            mount(<Color title="Test Color" color="#F0F0F0" timestamp="Mon Apr 11 2016 12:54:19 GMT-0700 (PDT)" />)
                .find('div.star-rating__stars')
                .childAt(2)
                .simulate('click')
        })

        it("Invokes onRate property", () => {
            mount(<Color title="Test Color" color="#F0F0F0" timestamp="Mon Apr 11 2016 12:54:19 GMT-0700 (PDT)" />)
                .find("button")
                .simulate('click')
        })
    });
});