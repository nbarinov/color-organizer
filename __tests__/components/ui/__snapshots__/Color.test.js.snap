// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Color /> UI Component Rendering for UI Renders correct properties 1`] = `"<section class=\\"color\\"><header class=\\"color__header\\"><h1 class=\\"color__title\\">Test Color</h1><button class=\\"color__remove\\" title=\\"remove color\\">x</button></header><div class=\\"color__block\\" style=\\"background-color:#f0f0f0\\"></div><footer class=\\"color__footer\\"><div class=\\"color__rating star-rating\\"><div class=\\"star-rating__stars\\"><div class=\\"star-rating__star star star--selected\\"></div><div class=\\"star-rating__star star star--selected\\"></div><div class=\\"star-rating__star star star--selected\\"></div><div class=\\"star-rating__star star\\"></div><div class=\\"star-rating__star star\\"></div></div><p class=\\"star-rating__total\\">3 of 5</p></div><div class=\\"color__timeago time-ago\\">4/11/2016</div></footer></section>"`;
